test/parallel/test-https-agent-secure-protocol.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+var assert = require('assert');
+var common = require('../common');
+
+if (!common.hasCrypto) {
+  common.skip('missing crypto');
+  return;
+}
+
+var https = require('https');
+var fs = require('fs');
+
+var options = {
+  key: fs.readFileSync(common.fixturesDir + '/keys/agent1-key.pem'),
+  cert: fs.readFileSync(common.fixturesDir + '/keys/agent1-cert.pem'),
+  ca:  fs.readFileSync(common.fixturesDir + '/keys/ca1-cert.pem')
+};
+
+
+var server = https.Server(options, function(req, res) {
+  res.writeHead(200);
+  res.end('hello world\n');
+});
+
+server.listen(0, function() {
+  const port = this.address().port;
+  var globalAgent = https.globalAgent;
+  globalAgent.keepAlive = true;
+  https.get({
+    path: '/',
+    port: port,
+    ca: options.ca,
+    rejectUnauthorized: true,
+    servername: 'agent1',
+    secureProtocol: 'SSLv23_method'
+  }, function(res) {
+    res.resume();
+    globalAgent.once('free', function () {
+      https.get({
+        path: '/',
+        port: port,
+        ca: options.ca,
+        rejectUnauthorized: true,
+        servername: 'agent1',
+        secureProtocol: 'TLSv1_method'
+      }, function(res) {
+        res.resume();
+        globalAgent.once('free', function () {
+          // Verify that two keep-alived connections are created
+          // due to the different secureProtocol settings:
+          var keys = Object.keys(globalAgent.freeSockets);
+          assert.equal(keys.length, 2);
+          assert.notEqual(keys[0].indexOf(':SSLv23_method'), -1);
+          assert.notEqual(keys[1].indexOf(':TLSv1_method'), -1);
```

Comment:
Can you use `assert.strictEqual` and `assert.notStrictEqual` here?  Also, instead of .indexOf() you can use .includes().